src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,841 @@ static DamageTable D_80AD2550 = {
+            return;
+
+        case 3:
+            this->actor.flags &= ~ACTOR_FLAG_1;
+            this->actionFunc = func_80AD16A8;
+            return;
+
+        default:
+            Actor_MarkForDeath(&this->actor);
+            return;
+    }
+}
+
+void EnOsn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOsn* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnOsn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    u32 pad1;
+    u16 pad2;
+    u32 sp34;
+    EnOsn* this = THIS;
+
+    sp34 = Flags_GetSwitch(globalCtx, 0);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveWithGravity(&this->actor);
+    SkelAnime_Update(&this->skelAnime);
+    if ((this->actor.params & 3) == 0) {
+        if (sp34 != 0) {
+            this->actor.flags |= ACTOR_FLAG_1;
+            func_80AD0830(this, globalCtx);
+            this->actor.draw = EnOsn_Draw;
+        } else {
+            this->actor.draw = NULL;
+            this->actor.flags &= ~ACTOR_FLAG_1;
+        }
+    }
+    func_800E9250(globalCtx, &this->actor, &this->unk_1D8, &this->unk_1DE, this->actor.focus.pos);
+}
+
+s32 EnOsn_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* actor,
+                           Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
+
+    if (this->unk_1F0 && limbIndex == 0xB) {
+        Matrix_InsertXRotation_s(this->unk_1D8.y, 1);
+    }
+    if ((this->unk_1EC == 9 || this->unk_1EC == 8) && limbIndex == 0xA) {
+        *dList = 0;
+    }
+    return 0;
+}
+
+void EnOsn_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
+    Vec3f sp30 = D_80AD2574;
+    Vec3s sp28 = D_80AD2580;
+
+    if (limbIndex == 0xB) {
+        Matrix_MultiplyVector3fByState(&sp30, &actor->focus.pos);
+    }
+    if (((this->unk_1EC == 0x11) || (this->unk_1EC == 0x15) || (this->unk_1EC == 0x16)) && (limbIndex == 6)) {
+        Matrix_StatePush();
+        Matrix_InsertTranslation(-400.0f, 1100.0f, -200.0f, 1);
+        Matrix_InsertXRotation_s(sp28.x, 1);
+        Matrix_RotateY(sp28.y, 1);
+        Matrix_InsertZRotation_s(sp28.z, 1);
+
+        gSPMatrix((*gfx)++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList((*gfx)++, &object_osn_DL_0192A0);
+        Matrix_StatePop();
+    }
+}
+
+void EnOsn_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    GraphicsContext* temp_s0;
+    EnOsn* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if (this->unk_1FC == 0xFF) {
+        func_8012C28C(globalCtx->state.gfxCtx);
+        if ((this->unk_1EC == 0xB) || (this->unk_1EC == 0xC) || (this->unk_1EC == 0x17) ||
+            (globalCtx->msgCtx.unk11F04 == 0x1FCA)) {
+            gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(D_80AD258C));
```

Comment:
Given that this draw function checks if unk_1FC is 0xFF to use OPA vs XLU, and passes unk_1EC as the last parameter in `gDPSetEnvColor` I'm pretty sure its the actor's alpha in a standard RGBA color.

You could probably rename that actor struct value at this point, although I wouldn't require it. 

(if you do dont forget to change it from hex to dec everywhere it gets used)